### PR TITLE
ZK-4175: replace XHR with the fetch() API to allow SSO redirect handling

### PR DESCRIPTION
**Pull request by plijyfes**

### CODE REVIEW

File: zk/src/archive/web/js/zk/au.js [hunk 1]
```diff
@@ -957,22 +948,22 @@ zAu.beforeSend = function (uri, req, dt) {
...
+	_respFailure: function (response, reqInf, rstatus) {
 		var eru = zAu._errURIs['' + rstatus];
 		if (typeof eru == 'string') {
 			zUtl.go(eru);
 			return true;
 		}
 
 		if (typeof zAu.ajaxErrorHandler == 'function') {
-			zAu.ajaxReqTries = zAu.ajaxErrorHandler(req, rstatus, req.statusText, zAu.ajaxReqTries);
+			zAu.ajaxReqTries = zAu.ajaxErrorHandler(response, rstatus, req.statusText, zAu.ajaxReqTries);
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/au.js:966]: response.statusText

File: zk/src/archive/web/js/zk/au.js [hunk 2]
```diff
@@ -1042,31 +1033,32 @@ zAu.beforeSend = function (uri, req, dt) {
...
 		} catch (e) {
 			if (e.name == 'SyntaxError') { //ZK-4199: handle json parse error
 				zAu.showError('FAILED_TO_PARSE_RESPONSE', e.message);
-				zk.debugLog(e.message + ', response text:\n' + req.responseText);
+				zk.debugLog(e.message + ', response text:\n' + rt);
 				return false;
 			}
 			throw e;
 		}
+
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/au.js:1061]: ?

Comment (scribetw) [inline zk/src/archive/web/js/zk/au.js:1061]: ?

File: zk/src/archive/web/js/zk/au.js [hunk 3]
```diff
@@ -1138,10 +1130,10 @@ zAu.beforeSend = function (uri, req, dt) {
 			setTimeout(ajaxReqResend2, timeout ? timeout : 0);
 		}
 	},
-	onResponseError: function (req, errCode) {
+	onResponseError: function (response, errCode) {
 		//$clone first since it might add or remove onError
 		for (var errs = _onErrs.$clone(), fn; fn = errs.shift();)
-			if (fn(req, errCode))
+			if (fn(response, errCode))
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/au.js:1136]: zAu.onError API changes

File: zk/src/archive/web/js/zk/ext/promise-polyfill.js [hunk 4]
```diff
@@ -0,0 +1,312 @@
+/* promise-polyfill.js.ts
```
Comment (plijyfes) [inline zk/src/archive/web/js/zk/ext/promise-polyfill.js:1]: .js

**Pull request merged**